plugins/languages/treesitter/treesitter.nix

```diff
@@ -6,210 +6,317 @@
-            (listOf str)
-          ];
-        default = "all";
-        description = "Either \"all\" or a list of languages";
-      };
+  # TODO introduced 2024-07-06: remove after 24.11
+  optionsRenamedToSettings = [
+    "ensureInstalled"
+    "ignoreInstall"
+    "parserInstallDir"
+    [
+      "incrementalSelection"
+      "enable"
+    ]
+    [
+      "incrementalSelection"
+      "keymaps"
+      "initSelection"
+      "nodeDecremental"
+    ]
+    [
+      "incrementalSelection"
+      "keymaps"
+      "initSelection"
+      "nodeIncremental"
+    ]
+    [
+      "incrementalSelection"
+      "keymaps"
+      "initSelection"
+      "scopeIncremental"
+    ]
+  ];
 
-      gccPackage = helpers.mkPackageOption {
-        default = if cfg.nixGrammars then null else pkgs.gcc;
-        description = ''
-          Which package (if any) to be added as the GCC compiler.
-          This is required to build grammars if you are not using `nixGrammars`.
-          To disable the installation of GCC, set this option to `null`.
-        '';
-      };
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "treesitter"
+      ];
+      settingsPath = basePluginPath ++ [ "settings" ];
+    in
+    [
+      (lib.mkRenamedOptionModule (basePluginPath ++ [ "customCaptures" ]) (
+        settingsPath
+        ++ [
+          "highlight"
+          "custom_captures"
+        ]
+      ))
+      (lib.mkRenamedOptionModule (basePluginPath ++ [ "disabledLanguages" ]) (
+        settingsPath
+        ++ [
+          "highlight"
+          "disable"
+        ]
+      ))
+      (lib.mkRenamedOptionModule (basePluginPath ++ [ "indent" ]) (
+        settingsPath
+        ++ [
+          "indent"
+          "enable"
+        ]
+      ))
+    ];
+
+  settingsOptions = {
+    auto_install = helpers.defaultNullOpts.mkBool false ''
+      Whether to automatically install missing parsers when entering a buffer.
+    '';
 
-      parserInstallDir = mkOption {
-        type = types.nullOr types.str;
-        default = if cfg.nixGrammars then null else "$XDG_DATA_HOME/nvim/treesitter";
-        description = ''
-          Location of the parsers to be installed by the plugin (only needed when nixGrammars is disabled).
-          This default might not work on your own install, please make sure that $XDG_DATA_HOME is set if you want to use the default. Otherwise, change it to something that will work for you!
+    highlight = {
+      additional_vim_regex_highlighting =
+        with helpers.nixvimTypes;
+        helpers.defaultNullOpts.mkNullableWithRaw (either bool (listOf (maybeRaw str))) false ''
+          Setting this to true will run `:h syntax` and tree-sitter at the same time. \
+          Set this to `true` if you depend on 'syntax' being enabled (like for indentation). \
+          Using this option may slow down your editor, and you may see some duplicate highlights. \
+          Instead of true it can also be a list of languages
```

Comment:
```suggestion
          Setting this to true will run `syntax` and tree-sitter at the same time. \
          Set this to `true` if you depend on 'syntax' being enabled (e.g. for indentation). \
          See `:h syntax`.

          Using this option may slow down your editor, and you may see some duplicate highlights. \
          Instead of true, it can also be a list of languages.
```